Source/Dafny/Verifier/Translator.cs

```diff
@@ -9416,7 +9416,7 @@ AND  Apply(f,h0,bxs) == Apply(f,h0,bxs)
       // Create the type constructor
       if (td is ClassDecl cl && cl.IsObjectTrait) {
         // the type constructor for "object" is in DafnyPrelude.bpl
-      } else if (td is TupleTypeDecl ttd && ttd.Dims == 2) {
+      } else if (td is TupleTypeDecl ttd && ttd.Dims == 2 && ttd.NonGhostDims == 2) {
```

Comment:
Should `ttd.Dims == 2` still be in here?

Reply:
Actually, shouldn't it be `ttd.Dims == 2 && ttd.NonGhostDims == 0`?
Also, the same test appears in one more place in `Translator.cs`, around line 1387.

Reply:
I think `ttd.Dims == 2 && ttd.NonGhostDims == 2` is correct. See [this message](https://github.com/dafny-lang/dafny/pull/1269#discussion_r680798493) in the other conversation.